src/ui/window.c

```diff
@@ -1464,9 +1460,7 @@ win_print_outgoing(ProfWin* window, const char* show_char, const char* const id,
     GDateTime* timestamp = g_date_time_new_now_local();
 
     const char* myjid = connection_get_fulljid();
-    if (replace_id) {
-        _win_correct(window, message, id, replace_id, myjid);
-    } else {
+    if (!replace_id || !_win_correct(window, message, id, replace_id, myjid)) {
```

Comment:
Why isn't `PREF_CORRECTION_ALLOW` checked here?

Reply:
It's checked priorly, in the cmd_correct. That's a prior mess, I didn't want to change the logic and set checking everywhere in cmd_function/replace function, as it's a design solution and it would require further negotiations etc.

Reply:
@jubalh, where the check for pref should be done? In cmd_correct it shows a good error, thus being more user-friendly. As for me, I am not sure that we should make this check in the internal function, since it's needed for some other functions (e.g. download progress)

Reply:
IMO that check doesn't necessarily only have to be once. If it's better to be displayed in the UI, please do it once where the error reporting is good, but also do it later in the code (maybe a second time) if it improves the code overall.

Reply:
Would be good to have it solved correctly but I agree that its not the point of your PR. It would be great to have it in a 2nd commit though. I'll have to leave now and will merge in the evening if all things are addressed and @sjaeckel gave approval as well. Thanks for the work everybody!